scene/gui/spin_box.cpp

```diff
@@ -68,7 +72,13 @@ void SpinBox::_update_text(bool p_keep_line_edit) {
 }
 
 void SpinBox::_text_submitted(const String &p_string) {
-	if (p_string.is_empty()) {
+	if (p_string.is_empty() || (update_on_text_changed && !p_string.begins_with(".") && p_string.ends_with("."))) {
+		return;
+	}
+
+	if (update_on_text_changed && p_string.begins_with(".")) {
+		line_edit->set_text("0.");
```

Comment:
Adding the `0.` automatically when `.` is the first character can be done with `update_on_text_changed` is false, if desired.